tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/cocosnet_evaluator.py

```diff
@@ -0,0 +1,411 @@
+"""
+Copyright (c) 2020 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from pathlib import Path
+import pickle
+from functools import partial
+from collections import OrderedDict
+import numpy as np
+
+from ..base_evaluator import BaseEvaluator
+from ..quantization_model_evaluator import create_dataset_attributes
+from ...utils import get_path, extract_image_representations
+from ...dataset import Dataset
+from ...launcher import create_launcher, DummyLauncher, InputFeeder
+from ...launcher.loaders import PickleLoader
+from ...logging import print_info, warning
+from ...metrics import MetricsExecutor
+from ...postprocessor import PostprocessingExecutor
+from ...preprocessor import PreprocessingExecutor, Preprocessor
+from ...adapters import create_adapter
+from ...config import ConfigError
+from ...data_readers import BaseReader, REQUIRES_ANNOTATIONS, DataRepresentation
+
+
+# pylint: disable=W0223
+class CocosnetEvaluator(BaseEvaluator):
+    def __init__(
+            self, launcher, reader, preprocessor, postprocessor, dataset, metric, model
+    ):
+        self.launcher = launcher
+        self.reader = reader
+        self.preprocessor_mask = [
```

Comment:
better to create preprocessing executor and provide all necessary preprocessing configuration instead of direct iteration